src/learning-catalogue/service/courseService.ts

```diff
@@ -0,0 +1,94 @@
+import {Course} from '../model/course'
+import {CourseFactory} from '../model/factory/courseFactory'
+import {DefaultPageResults} from '../model/defaultPageResults'
+import {AxiosInstance} from 'axios'
+import {LearningCatalogueConfig} from '../learningCatalogueConfig'
+
+export class CourseService {
+	private _http: AxiosInstance
+	private _config: LearningCatalogueConfig
+	private _courseFactory: CourseFactory
+
+	constructor(http: AxiosInstance, config: LearningCatalogueConfig) {
+		this._http = http
+		this._config = config
+		this._courseFactory = new CourseFactory()
+	}
+
+	async listAll(
+		page: number = 0,
+		size: number = 10
+	): Promise<DefaultPageResults<Course>> {
+		try {
+			const response = await this._http.get(
```

Comment:
Course and Module service this piece of code is repeating, only change is the url that is being passed.
Extract a protected method maybe?
Extract auth injection on a higher level also maybe so it is being done in one place?